include/sys/mutex.h

```diff
@@ -78,16 +80,24 @@ spl_mutex_set_type(kmutex_t *mp, kmutex_type_t type)
 {
 	u32 t = 0;
 
-	ASSERT((type == MUTEX_DEFAULT) || (type == MUTEX_NOLOCKDEP));
+	ASSERT((type == MUTEX_DEFAULT) || (type == MUTEX_NOLOCKDEP) ||
+	    (type == MUTEX_NODEBUG));
 
 	/*
 	 * Mutex m_flags flag/counter bits:
 	 * bit 31 : NOLOCKDEP
-	 * bits [30 - 0]: mutex_exit counter (m_flags & ~MUTEX_FLAG_MASK)
+	 * bit 30 : NODEBUG
+	 * bits [29 - 0]: mutex_exit counter (m_flags & ~MUTEX_FLAG_MASK)
 	 */
-
-	if (type == MUTEX_NOLOCKDEP)
-		t |= MUTEX_FLAG_NOLOCKDEP;
+	switch (type) {
+		case MUTEX_NODEBUG:
+			t |= MUTEX_FLAG_NODEBUG;
```

Comment:
Yes, I initially intended for NODEBUG to rule out all other checks.
Or should we add this check as a separate category, e.g. MUTEX_LOCKTRACKING?